utils/proxy/core.py

```diff
@@ -187,6 +201,52 @@ def start_proxy(state) -> None:
     asyncio.run_coroutine_threadsafe(proxy.run(), loop)
 
 
+def start_backend_proxy(state=None) -> None:
```

Comment:
Would it be possible to use the existing thread for the proxy, using `proxies` argument on request module ?